controllers/IssueController.js

```diff
@@ -424,16 +425,39 @@ const addIssueController = async function (req, res) {
     if (errors) {
       res.status(400).send(new Error("BAD_REQUEST", errors));
     } else {
-      const issue = await createIssue(
+      const isNSFW = await hasNSFWText(title, description, scope);
+
+      const { id: issueId } = await createIssue(
         title,
         description,
         section,
         images,
         department,
         scope.toUpperCase(),
+        isNSFW,
         id
       );
-      if (images.length > 0) await updateImageIssueId(images, issue.id);
+
+      if (images.length > 0) await updateImageIssueId(images, issueId);
+
+      if (isNSFW) {
+        const user = await getUserById(id);
+
+        user.violations.push(issueId);
+        if (user.violations.length >= USER_VIOLATIONS_THRESHOLD) {
+          user.isDisabled = true;
```

Comment:
Here isn't the used logged out ? Or the `jwt` is invalidated?

Reply:
This is the controller function. Sign-in status and JWT validation is all done at in the routes.